pkg/crc/machine/bundle/copier.go

```diff
@@ -0,0 +1,166 @@
+package bundle
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/code-ready/crc/pkg/compress"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	crcos "github.com/code-ready/crc/pkg/os"
+	godeepcopy "github.com/jinzhu/copier"
+)
+
+type Copier struct {
+	srcBundle    *CrcBundleInfo
+	copiedBundle CrcBundleInfo
+}
+
+func NewCopier(srcBundle *CrcBundleInfo, basePath string, customBundleName string) (*Copier, error) {
+	var copier Copier
+
+	bundlePath := filepath.Join(basePath, customBundleName)
+	if err := os.Mkdir(bundlePath, 0775); err != nil {
+		return nil, err
+	}
+	if err := godeepcopy.Copy(&copier.copiedBundle, srcBundle); err != nil {
+		return nil, err
+	}
+
+	copier.copiedBundle.bundleName = customBundleName
+	copier.copiedBundle.Type = "custom"
+	copier.srcBundle = srcBundle
+	copier.copiedBundle.cachedPath = bundlePath
+
+	return &copier, nil
+}
+
+func (copier *Copier) Cleanup() error {
+	return os.RemoveAll(copier.copiedBundle.cachedPath)
+}
+
+func (copier *Copier) resolvePath(filename string) string {
+	return copier.copiedBundle.resolvePath(filename)
+}
+
+func (copier *Copier) CopyPrivateSSHKey(srcPath string) error {
+	sshKeyFileName := filepath.Base(copier.srcBundle.GetSSHKeyPath())
+	destPath := copier.resolvePath(sshKeyFileName)
+	if err := crcos.CopyFileContents(srcPath, destPath, 0400); err != nil {
+		return err
+	}
+	// copier.srcBundle.ClusterInfo.SSHPrivateKeyFile = sshKeyFileName
```

Comment:
you need to update the filename in `copier.copiedBundle`

Reply:
@cfergeau yes but I wanted to have discussion here why it is even required because it is the fileName which we also want same as what is in the source bundle so why need to update it because copiedBundle is anyway a deep copy of source right?

Reply:
The source bundle uses `copier.srcBundle.GetSSHKeyPath()` as its ssh key filename
The destination bundle uses `srcPath` as its ssh key filename
Are both names going to be the same (now and in the future)? We can't know, so if we add a ssh key named `srcPath` in the  destination bundle, then we need to update the corresponding field.

Reply:
`SSHPrivateKeyFile` is still not updated